java/cdk/src/main/java/sleeper/cdk/stack/PartitionSplittingStack.java

```diff
@@ -68,51 +73,96 @@
-                                   BuiltJars jars,
-                                   CoreStacks coreStacks,
-                                   Topic topic) {
+            String id,
+            InstanceProperties instanceProperties,
+            BuiltJars jars,
+            CoreStacks coreStacks,
+            Topic topic) {
         super(scope, id);
 
         // Jars bucket
         IBucket jarsBucket = Bucket.fromBucketName(this, "JarsBucket", jars.bucketName());
 
+        // Create queue for batching tables
+        QueueAndDlq batchQueueAndDlq = createBatchQueues(instanceProperties);
+        this.partitionSplittingBatchQueue = batchQueueAndDlq.queue;
         // Create queue for partition splitting job definitions
-        this.dlQueue = Queue.Builder
+        QueueAndDlq jobQueueAndDlq = createJobQueues(instanceProperties, topic);
+        this.partitionSplittingJobQueue = jobQueueAndDlq.queue;
+        this.partitionSplittingJobDlq = jobQueueAndDlq.dlq;
+
+        // Partition splitting code
+        LambdaCode splitterJar = jars.lambdaCode(BuiltJar.PARTITION_SPLITTER, jarsBucket);
+        Map<String, String> environmentVariables = Utils.createDefaultEnvironment(instanceProperties);
+
+        // Lambda to batch tables and put requests on the batch SQS queue, to be consumed by FindPartitionsToSplit
+        createTriggerFunction(instanceProperties, splitterJar, coreStacks, environmentVariables);
+
+        // Lambda to look for partitions that need splitting (for each partition that
+        // needs splitting it puts a definition of the splitting job onto a queue)
+        createFindPartitionsToSplitFunction(instanceProperties, splitterJar, coreStacks, environmentVariables);
+
+        // Lambda to split partitions (triggered by partition splitting job
+        // arriving on partitionSplittingQueue)
+        createSplitPartitionFunction(instanceProperties, splitterJar, coreStacks, environmentVariables);
+
+        Utils.addStackTagIfSet(this, instanceProperties);
+    }
+
+    private QueueAndDlq createBatchQueues(InstanceProperties instanceProperties) {
+        // Create queue for batching tables
+        Queue partitionSplittingBatchDlq = Queue.Builder
+                .create(this, "PartitionSplittingBatchDeadLetterQueue")
+                .queueName(instanceProperties.get(ID) + "-PartitionSplittingBatchDLQueue")
+                .build();
+        Queue partitionSplittingBatchQueue = Queue.Builder
+                .create(this, "PartitionSplittingBatchQueue")
+                .queueName(instanceProperties.get(ID) + "-PartitionSplittingBatchQueue")
+                .deadLetterQueue(
+                        DeadLetterQueue.builder()
+                                .maxReceiveCount(1)
+                                .queue(partitionSplittingBatchDlq)
+                                .build())
+                .visibilityTimeout(Duration.seconds(instanceProperties.getInt(SPLIT_PARTITIONS_TIMEOUT_IN_SECONDS)))
```

Comment:
This timeout needs to agree with the timeout for the find partitions to split lambda, since that's the one that will process messages from this queue.